src/custom hook/useFetch.js

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect } from "react";
+
+import { useLocation, useParams, useSearchParams } from "react-router-dom";
+
+function useFetch() {
+  const [data, setData] = useState();
+  const params = useParams();
+  const { pathname, search } = useLocation();
+  const [searchParams] = useSearchParams();
+  const { genre, sortBy } = Object.fromEntries([...searchParams]);
+
+  const fetchData = (params) => {
+    return fetch("http://localhost:4000/movies" + params)
+      .then((res) => res.json())
+      .then((json) => setData(json.data));
+  };
+
+  useEffect(() => {
+    if (
+      (!search && location.pathname === "/search") ||
+      location.pathname === "/search/"
+    ) {
+      fetchData("?sortBy=vote_average&sortOrder=desc");
+    } else {
+      if (genre) {
+        fetchData(`?searchBy=genres&filter=${genre}`);
+      } else if (params.movieName && sortBy) {
+        fetchData(
+          `?sortBy=${sortBy}&sortOrder=desc&search=${params.movieName
+            .split(" ")
+            .join("%20")}&searchBy=title&limit=6`
```

Comment:
creating constant with proper name ?

Reply:
Also check other places please

Reply:
/search was changed to defined constant variable in everywhere. Fixed